scripts/ci/selective_tests.sh

```diff
@@ -125,18 +182,60 @@ function count_changed_files() {
     echo "${count_changed_files}"
 }
 
-function run_all_tests_when_push_or_schedule() {
+function run_everything_when_push_or_schedule() {
     if [[ ${GITHUB_EVENT_NAME} == "push" || ${GITHUB_EVENT_NAME} == "schedule" ]]; then
         echo
         echo "Always run all tests in case of push/schedule events"
         echo
-        set_outputs_run_all_tests
-        exit
+        set_outputs_run_everything_and_exit
+    fi
+}
+
+function check_if_api_tests_should_be_run() {
+    local api_triggering_patterns=(
+        "^airflow/api"
```

Comment:
This should include airflow/models/ too I think -- at lot of query logic the API relies on lives in the model code.

Reply:
The intention here is, that if any "core/other" file changes (including models), then always all the tests are run (so all API, unit etc. tests should be always run when those change). So there is no need to add them here specifically. If "models" change, then everything else - including providers, cli, etc. are impacted and all tests should be run.

I will double-check if that's the case. 

This selection is only really used, in case ONLY API files change - then ONLY API tests should be run. This work for sure for the regular "unit" API tests, but I will double check if that's the case for the client generation

Reply:
Ah got it - didn't get that from reading the diff but I didn't try to follow all the logic exactly

Reply:
I will try to describe it in some way that will be easy to read and grasp :). It's difficult though ...